.github/workflows/dev.yml

```diff
@@ -0,0 +1,22 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  pull_request:
```

Comment:
On push also should be here.

The difference between pull request and push into dev should be the following:

- pull request - just build the project `mvn clean install`
- push into `dev` - upload to the Github packages - `mvn clean deploy`

Reply:
To achieve it we can either use conditions or create a different workflow.